### PR TITLE
fix: update AnnualProvisions once per year

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1870

Note: this wasn't caught by the existing `TestAnnualProvisions` unit test because the test app used `stake` as its bond denomination but the mint module uses `utia` as the default bond denomination. So this PR also updates the test app to use `utia` as the default bond denomination. After this change, new tokens that get emitted due to inflation increase the total supply of `utia` which will impact the annual provisions calculation.

### CODE REVIEW

File: test/util/test_app.go [hunk 1]
```diff
@@ -250,8 +250,10 @@ func genesisStateWithValSet(
 
 	}
 	// set validators and delegations
-	stakingGenesis := stakingtypes.NewGenesisState(stakingtypes.DefaultParams(), validators, delegations)
-	genesisState[stakingtypes.ModuleName] = app.AppCodec().MustMarshalJSON(stakingGenesis)
+	params := stakingtypes.DefaultParams()
+	params.BondDenom = app.BondDenom
```
Comment (rootulp) [inline test/util/test_app.go:254]: this line makes the test app's staking module use `utia` instead of `stake`

**Review by evan-forbes — APPROVED** [review]
really nice catch!

File: x/mint/abci.go [hunk 2]
```diff
@@ -35,7 +35,9 @@ func maybeSetGenesisTime(ctx sdk.Context, k keeper.Keeper) {
 func maybeUpdateMinter(ctx sdk.Context, k keeper.Keeper) {
 	minter := k.GetMinter(ctx)
 	newInflationRate := minter.CalculateInflationRate(ctx)
-	if newInflationRate == minter.InflationRate {
+
+	isNonZeroAnnualProvisions := !minter.AnnualProvisions.IsZero()
+	if newInflationRate.Equal(minter.InflationRate) && isNonZeroAnnualProvisions {
 		// The minter's InflationRate AnnualProvisions already reflect the
 		// values for this year. Exit early because we don't need to update
 		// them.
```
Comment (evan-forbes) [inline x/mint/abci.go:43]: the good news is this appears to not break state or the result of the inflation, but previous to this change, this code was literally never getting hit, so we would always end up updating inflation. That might actually still change the apphash even if the state is not changing, depending on if this is causing an insertion to the iavl tree

**Comment by evan-forbes** [comment]
@rootulp while we're fixing this, can we also only call `SetMinter` only once here? I don't think there is any benefit to calling this more than once, if anything only weirdness could occur

https://github.com/celestiaorg/celestia-app/blob/348f795d8635324db4cca0367b599b5edc5c2fdb/x/mint/abci.go#L44-L49

**Comment by rootulp** [comment]
>  can we also only call `SetMinter` only once here?

Sure done in https://github.com/celestiaorg/celestia-app/pull/1871/commits/2d7dab5037f45c63e8095bcf9ec1c461eab9a794

**Review by evan-forbes — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**